docs/reference/built-in-readers.md

```diff
@@ -0,0 +1,88 @@
+```
+
+### The mapping.json file
+
+This file is designed to let you fill in the requirements of a NeXus Application Definition without writing any code. If you already have data in the formats listed above, you just need to use this mapping file to help the dataconverter pick your data correctly.
+
+The mapping files will always be based on the Template the dataconverter generates. See above on how to generate a mapping file.
+The right hand side values of the Template keys are what you can modify.
+
+Here are the three different ways you can fill the right hand side of the Template keys:
+* Write the nested path in your datafile. This is indicated by a leading `/` before the word `entry` to make `/entry/data/current_295C` below.
+Example:
+
+```json
+  "/ENTRY[entry]/DATA[data]/current_295C": "/entry/data/current_295C",
+  "/ENTRY[entry]/NXODD_name/posint_value": "/a_level_down/another_level_down/posint_value",
+```
+
+* Write the values directly in the mapping file for missing data from your data file.
+
+```json
+  "/ENTRY[entry]/PROCESS[process]/program": "Bluesky",
+  "/ENTRY[entry]/PROCESS[process]/program/@version": "1.6.7"
+```
+
+* Write JSON objects with a link key. This follows the same link mechanism that the dataconverter implements. In the context of this reader, you can only use external links to your data files. In the example below, `current.nxs` is an already existing HDF5 file that we link to in our new NeXus file without copying over the data. The format is as follows:
+`"link": "<filename>:<path_in_file>"`
+Note: This only works for HDF5 files currently.
+
+```json
+  "/ENTRY[entry]/DATA[data]/current_295C": {"link": "current.nxs:/entry/data/current_295C"},
+  "/ENTRY[entry]/DATA[data]/current_300C": {"link": "current.nxs:/entry/data/current_300C"},
+```
+
+### Examples
+
+#### Basic mappings
+
+There are some example files you can use:
+
+[data.mapping.json](https://github.com/FAIRmat-NFDI/pynxtools/tree/master/tests/data/dataconverter/readers/json_map/data.mapping.json)
+
+[data.json](https://github.com/FAIRmat-NFDI/pynxtools/tree/master/tests/data/dataconverter/readers/json_map/data.json)
+
+```console
+user@box:~$ dataconverter --nxdl NXtest data.json --mapping data.mapping.json
+```
+
+#### Example with HDF5 files
+You can find example data files for using the mapping with HDF5 files at [`examples/json_map`](https://github.com/FAIRmat-NFDI/pynxtools/tree/master/examples/examples/json_map/).
+
+The example can be run by calling
+
+
+```console
+user@box:~$ dataconverter --nxdl nxdl any_data.hdf5 --mapping my_custom_map.mapping.json
+```
+
+
+## The [YamlJsonReader](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/readers/json_yml/reader.py)
+!!! info "Work in progress"
+
+## Installation
+
+Each of the built-in reader comes with the main `pynxtools` package, therefore they are avaible after pip installation:

```

Comment:
"Each of the built-in readers are shipped/installed with the main `pynxtools` package. Those readers are available ..."

Reply:
changed